unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/Font/AssetPromise_Font.cs

```diff
@@ -40,61 +30,54 @@ public AssetPromise_Font(string src)
 
         protected override void OnCancelLoading()
         {
-            CoroutineStarter.Stop(fontCoroutine);
+            CleanCT();
         }
 
         public override void Cleanup()
         {
             base.Cleanup();
-            CoroutineStarter.Stop(fontCoroutine);
+            CleanCT();
         }
 
         protected override void OnLoad(Action OnSuccess, Action<Exception> OnFail)
         {
-            if (fontsMapping.TryGetValue(src, out string fontResourceName))
-            {
-                fontCoroutine = CoroutineStarter.Start(GetFontFromResources(OnSuccess, OnFail, fontResourceName));
-            }
-            else
+            cancellationTokenSource = new CancellationTokenSource();
+
+            async UniTaskVoid LaunchRequest()
             {
-                OnFail?.Invoke(new Exception("Font doesn't correspond with any know font"));
+                FontResponse result = await fontResolver.Ref.GetFontAsync(permittedSources, src, cancellationTokenSource.Token);
+
+                if (result.IsSuccess)
+                {
+                    FontSuccessResponse successResponse = result.GetSuccessResponse();
+                    asset.font = successResponse.Font;
+                    OnSuccess?.Invoke();
+                }
+                else { OnFail?.Invoke(result.GetFailResponse().Exception); }
             }
+
+            LaunchRequest().Forget();
```

Comment:
I wonder if it makes sense to relaunch the flow if it was already launched? Consider applying
```
if (cancellationTokenSource != null)
                return;
```
What do you think?